client/src/components/Leaflet.js

```diff
@@ -242,6 +271,83 @@ const Leaflet = ({
     widthPropUnchanged
   ])
 
+  const { data } = API.useApiQuery(allLocationsQuery)
+
+  if (
+    data?.locationList?.list?.length &&
+    map &&
+    layerControl &&
+    !allLocationsLayer
+  ) {
+    const allMarkers = data.locationList.list
+      .filter(loc => Location.hasCoordinates(loc))
+      .map(location => {
+        const popupContent = document.createElement("div")
+        popupContent.setAttribute("style", "width: 300px;text-align: center")
+
+        return new Marker([location.lat, location.lng], {
+          icon: icon2,
+          draggable: false,
+          autoPan: false,
+          id: location.uuid
+        })
+          .bindPopup(popupContent)
+          .on("popupopen", e => {
+            // TODO LinkTo component will be utilized here to provide routing
+            ReactDOM.render(
+              <>
+                <b>{location.name}</b> @{" "}
+                <GeoLocation lat={location.lat} lng={location.lng} />
+              </>,
+              e.popup.getContent()
+            )
+          })
+      })
+
+    const locationsLayer = new MarkerClusterGroup({
+      iconCreateFunction: function(cluster) {
+        return new DivIcon({
+          className: "all-locations-marker-cluster-icon-container",
+          html: `
+              <img src="${MARKER_FLAG_BLUE}" class="alm-cluster-icon" alt="" />
+              <div class="alm-cluster-text">${cluster.getChildCount()}</div>
+          `
+        })
+      }
+    })
+      .addLayers(allMarkers)
+      .on("add", () => {
+        if (markers?.length && markers.some(m => m.onMove)) {
+          return // While editing a location don't change zoom levels
+        }
+
+        map.fitBounds(locationsLayer.getBounds())
```

Comment:
In most scenarios, we advise in locations across an entire country, or even several continents. fitting all locations on screen would be equivalent to fitting all available imagery on screen - basically it will always zoom out really a lot.
I suggest to treat the all locations layer as "background imagery" layer and to not auto fit to screen unless it is done explicitly.